test/utils/int/providerRemote.ts

```diff
@@ -20,3 +27,50 @@ export async function clearStoredNotifications(windowIdentity: Identity): Promis
         });
     }, windowIdentity);
 }
+
+/**
+ * Closes all of the service providers child windows and then restarts the main window after a provided amount of time.
+ * Resolves once the service is fully initialized.
+ * This will not reopen any child windows unless the provider automatically does so.
+ */
+export async function restartProvider(snoozeTime: number = 0): Promise<void> {
+    const providerApp = fin.Application.wrapSync(serviceIdentity);
+    const providerAppWindow = await providerApp.getWindow();
+    await providerApp.getChildWindows().then(children => children.forEach(win => win.close()));
+    await providerAppWindow.navigate('about:blank');
+
+    // A small delay is needed in order for OpenFin to handle the navigateBack call successfully.
+    // If a snoozeTime is less than the minimum we enforce the minimum otherwise let it be
+    await delay(Math.max(snoozeTime, Duration.NAVIGATE_BACK));
```

Comment:
Discovered the puppeteer docs the other day and theres a bunch of stuff we could be using for all this type of stuff like [`page.waitForNavigation([options])` ](https://pptr.dev/#?product=Puppeteer&version=v1.20.0&show=api-pagewaitfornavigationoptions)

Reply:
Wow, there is a lot of cool stuff in there we could / should use. We should have a good look over it and consider for what we want to keep in openfin vs what we can use in puppeteer. I would argue though that is beyond this ticket.